src/key.rs

```diff
@@ -268,54 +267,87 @@ where
     values: Vec<Option<RedisString>>,
 }
 
-impl<'a, T> HMGetResult<'a, T>
+pub struct HMGetIter<'a, T>
```

Comment:
Does `HMGetIter` need to be `pub`? Isn't it accessed only via the iterator?

Reply:
Yes, precisely _because_ it is accessed via the iterator:

```
error[E0446]: private type `key::HMGetIter<'a, T>` in public interface
   --> src/key.rs:308:5
    |
270 | struct HMGetIter<'a, T>
    | - `key::HMGetIter<'a, T>` declared as private
...
308 |     type IntoIter = HMGetIter<'a, T>;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type
```

Reply:
👍